crates/engine-api/src/client.rs

```diff
@@ -68,15 +71,21 @@ impl EngineApiClient {
     }
 
     async fn get<T: DeserializeOwned>(&self, uri: Uri) -> Result<T> {
+        let req = Request::builder()
+            .method(Method::GET)
+            .uri(uri)
+            .body(Either::Right(Empty::<Bytes>::new()))
+            .map_err(|err| anyhow!("Error building the request. Reason: {}", err))?;
```

Comment:
Super nit, something to handle in a separate PR: I think it would be better to use `thiserror` and return specific errors in such cases, especially since it's a client crate.